packages/pricing/src/repositories/price-rule.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  Context,
+  CreatePriceRuleDTO,
+  DAL,
+  UpdatePriceRuleDTO,
+} from "@medusajs/types"
+import { DALUtils, MedusaError } from "@medusajs/utils"
+import {
+  LoadStrategy,
+  FilterQuery as MikroFilterQuery,
+  FindOptions as MikroOptions,
+} from "@mikro-orm/core"
+import { PriceRule, PriceSet, PriceSetMoneyAmount, RuleType } from "@models"
+
+import { SqlEntityManager } from "@mikro-orm/postgresql"
+
+export class PriceRuleRepository extends DALUtils.MikroOrmBaseRepository {
+  protected readonly manager_: SqlEntityManager
+
+  constructor({ manager }: { manager: SqlEntityManager }) {
+    // @ts-ignore
+    // eslint-disable-next-line prefer-rest-params
+    super(...arguments)
+    this.manager_ = manager
+  }
+
+  async find(
+    findOptions: DAL.FindOptions<PriceRule> = { where: {} },
+    context: Context = {}
+  ): Promise<PriceRule[]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+
+    const findOptions_ = { ...findOptions }
+    findOptions_.options ??= {}
+
+    Object.assign(findOptions_.options, {
+      strategy: LoadStrategy.SELECT_IN,
+    })
+
+    return await manager.find(
+      PriceRule,
+      findOptions_.where as MikroFilterQuery<PriceRule>,
+      findOptions_.options as MikroOptions<PriceRule>
+    )
+  }
+
+  async findAndCount(
+    findOptions: DAL.FindOptions<PriceRule> = { where: {} },
+    context: Context = {}
+  ): Promise<[PriceRule[], number]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+
+    const findOptions_ = { ...findOptions }
+    findOptions_.options ??= {}
+
+    Object.assign(findOptions_.options, {
+      strategy: LoadStrategy.SELECT_IN,
+    })
+
+    return await manager.findAndCount(
+      PriceRule,
+      findOptions_.where as MikroFilterQuery<PriceRule>,
+      findOptions_.options as MikroOptions<PriceRule>
+    )
+  }
+
+  async delete(ids: string[], context: Context = {}): Promise<void> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+    await manager.nativeDelete(PriceRule, { id: { $in: ids } }, {})
+  }
+
+  async create(
+    data: CreatePriceRuleDTO[],
+    context: Context = {}
+  ): Promise<PriceRule[]> {
+    const manager: SqlEntityManager =
+      this.getActiveManager<SqlEntityManager>(context)
+
+    const toCreate = await Promise.all(data.map(async (ruleData) => {
+      const ruleDataClone = { ...ruleData } as unknown as PriceRule
+      ruleDataClone.rule_type = manager.getReference(
+        RuleType,
+        ruleData.rule_type_id
+      )
+
+      ruleDataClone.price_set = manager.getReference(
+        PriceSet,
+        ruleData.price_set_id
+      )
+
+      ruleDataClone.price_set_money_amount = await manager.findOne(
+        PriceSetMoneyAmount,
+        { id: ruleData.price_set_money_amount_id }
+      ) as PriceSetMoneyAmount
+
+      return ruleDataClone
+    }))
```

Comment:
todo(non blocking): I think it would be better to map first to create the clones, then findAll price set money amount and attach them on the values of the map in order to reduce to only one query instead of X

Reply:
fair warning, this will be changed in the calculate pricing PR. None of this will be happening in that.

Reply:
there's a much neater trick with mikro-orm that Riqwan pointed out to me after this pr was opened. You don't need references or anything: 

```
      ruleDataClone.rule_type = ruleData.rule_type_id

      ruleDataClone.price_set = ruleData.price_set_id

      ruleDataClone.price_set_money_amount = ruleData.price_set_money_amount_id 
```